libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1712,12 +1712,34 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 		);
 	}
 	else if (possibleMembers.size() > 1)
-		m_errorReporter.fatalTypeError(
-			_memberAccess.location(),
-			"Member \"" + memberName + "\" not unique "
-			"after argument-dependent lookup in " + exprType->toString() +
-			(memberName == "value" ? " - did you forget the \"payable\" modifier?" : "")
-		);
+	{
+		// Remove builtins (i.e. not having a declaration) first
+		for (auto it = possibleMembers.begin(); it != possibleMembers.end();)
+			if (
+				(
+					// Overloaded functions without declaration (e.g. transfer(), send(), call(), etc.)
+					it->type->category() == Type::Category::Function &&
+					!dynamic_cast<FunctionType const&>(*it->type).hasDeclaration()
```

Comment:
This  looks rather hacky. What about removing the members as part of the type instead of the overload resolution?
I would like `var  f = x.balance; f();` to work in the same way as `x.balance();`. If you need access to the account balance, you can use `address(x).balance`.

Reply:
It is very hacky, should be removed entirely at the breaking release when a contract isn't an child of address.

Do you have a non-hacky solution for keeping the members in non-conflicting cases?